### DISCUSSION TITLE
CCCV operating conditions

### DISCUSSION THREAD

**Discussion by balaji3112** [pybamm-team/liionpack#238] [Q&A]
I am trying to run an experiment that involves a constant current discharge followed by a constant voltage charging but, I'm receiving the following error:

    self.protocol = lp.generate_protocol_from_experiment(experiment, flatten=True)
  File "C:\Users\Welcome\.virtualenvs\Python_model\lib\site-packages\liionpack\protocols.py", line 32, in generate_protocol_from_experiment
    raise ValueError("Only constant current operations are supported")
ValueError: Only constant current operations are supported

Is it not possible to run CCCV in liionpack?

**Comment by TomTranter** [comment]
At this time only current operations are supported

**Comment by balaji3112** [comment]
Is it possible to simulate CCCV using only Pybamm? It would be helpful if you could guide me or suggest alternative methods to liionpack. Thank you very much.

**Reply by philgravy** [reply]
Hi balaji3112, I'm experiencing the exact same issue as yours. I'm looking to fully charge and discharge batteries and liionpack doesn't seem to support CC/CV related operations. I'm curious if you were ever able to resolve this. Thank you

**Comment by TomTranter** [comment]
At the moment the liionpack solvers are different to PyBaMM and only current is supported. It wouldn't be too difficult to add voltage conditions in liionpack but I haven't done this yet because a better solution I think is to rewrite the circuit problem to get solved alongside the electrochemistry problems and solve everything in PyBaMM. Liionpack will still exist for defining the circuits but much of the underlying code would get merged into PyBaMM. That's our plan for the next year or so. In the mean-time I don't want to spend too much time adding new features to liionpack which will eventually get replaced. I hope you understand.

**Reply by balaji3112** [reply]
Yes, I understand. Many thanks
